### PR TITLE
Adding ability to switch between StaggeredGridLayout and GridLayout

### PR DESCRIPTION
**Pull request by patbeagan1**
Squashed commit of the following:

commit a78b11a19fd27c6eecd492cbf294c6ad36daabff
Author: Patrick Beagan <pbeagan@wayfair.com>
Date:   Wed Mar 22 13:46:58 2017 -0400

    adding in more testing

commit 6a3b7dca2a00d20cbe1775f4c87bbc5d0861c832
Author: Patrick Beagan <pbeagan@wayfair.com>
Date:   Mon Mar 20 16:13:25 2017 -0400

    fixing lint

commit 39fffacb70cf9d7e653785188b78356a015928b4
Author: Patrick Beagan <pbeagan@wayfair.com>
Date:   Mon Mar 20 11:08:57 2017 -0400

    adding in testing and comments

commit cfe22b10fbffa1a9867e4da42f59d5883692628c
Author: Patrick Beagan <pbeagan@wayfair.com>
Date:   Fri Mar 17 16:37:50 2017 -0400

    adding in the layout changes

### CODE REVIEW

File: BrickKit/bricks/src/androidTest/java/com/wayfair/brickkit/BrickRecyclerAdapterTest.java [hunk 1]
```diff
@@ -558,4 +558,5 @@ public void testGetSectionFooterLastElement() {
     public void testGetSectionFooterBadPosition() {
         assertNull(adapter.getSectionFooter(-1));
     }
+
```
Comment (thusson13) [inline BrickKit/bricks/src/androidTest/java/com/wayfair/brickkit/BrickRecyclerAdapterTest.java]: probably can remove this rogue new line

File: BrickKit/bricks/src/main/java/com/wayfair/brickkit/brick/BaseBrick.java [hunk 2]
```diff
@@ -164,6 +165,26 @@ public void setStickyScrollMode(@StickyScrollMode int stickyScrollMode) {
     }
 
     /**
+     * Whether the brick should take up the width or height of the fragment
```
Comment (thusson13) [inline BrickKit/bricks/src/main/java/com/wayfair/brickkit/brick/BaseBrick.java]: can full width not be calculated somewhere else using the span size of the brick and span size of the brickdatamanager?

File: BrickKit/app/src/main/java/com/wayfair/brickkitdemo/StaggeredInfiniteScrollBrickFragment.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+import com.wayfair.brickkit.size.SimpleBrickSize;
+
+/**
+ * Example fragment which loads more bricks when scrolling to the bottom.
+ * <p>
+ * This fragment takes advantage of the {@link OnReachedItemAtPosition} which calls back when
+ * items are bound in the adapter.
+ */
+public class StaggeredInfiniteScrollBrickFragment extends BrickFragment {
+    private static final int HALF = 120;
```
Comment (thusson13) [inline BrickKit/app/src/main/java/com/wayfair/brickkitdemo/StaggeredInfiniteScrollBrickFragment.java]: not used anymore

File: BrickKit/bricks/src/main/java/com/wayfair/brickkit/brick/BaseBrick.java [hunk 4]
```diff
@@ -31,6 +31,7 @@
     private boolean isOnRightWall;
     @StickyScrollMode
     private int stickyScrollMode = StickyScrollMode.SHOW_ON_SCROLL;
+    private boolean fullSize;
```
Comment (thusson13) [inline BrickKit/bricks/src/main/java/com/wayfair/brickkit/brick/BaseBrick.java]: not used?

File: BrickKit/bricks/src/androidTest/java/com/wayfair/brickkit/BrickRecyclerAdapterTest.java (hunk 1 above)
Comment (patbeagan1) [inline BrickKit/bricks/src/androidTest/java/com/wayfair/brickkit/BrickRecyclerAdapterTest.java]: line has been taken out

File: BrickKit/bricks/src/main/java/com/wayfair/brickkit/brick/BaseBrick.java (hunk 2 above)
Comment (patbeagan1) [inline BrickKit/bricks/src/main/java/com/wayfair/brickkit/brick/BaseBrick.java]: revised to use the existing brick spansize

File: BrickKit/app/src/main/java/com/wayfair/brickkitdemo/StaggeredInfiniteScrollBrickFragment.java (hunk 3 above)
Comment (patbeagan1) [inline BrickKit/app/src/main/java/com/wayfair/brickkitdemo/StaggeredInfiniteScrollBrickFragment.java]: taken out

File: BrickKit/bricks/src/main/java/com/wayfair/brickkit/brick/BaseBrick.java (hunk 4 above)
Comment (patbeagan1) [inline BrickKit/bricks/src/main/java/com/wayfair/brickkit/brick/BaseBrick.java]: taken out

**Pull request merged**